pkg/util/k8sutil/pod_util.go

```diff
@@ -93,6 +93,8 @@ func applyPodPolicy(clusterName string, pod *v1.Pod, policy *api.PodPolicy) {
 		return
 	}
 
+	pod.Spec.ServiceAccountName = policy.ServiceAccountName
```

Comment:
You should probably check to make sure that this is filled before adding it to the pod spec